validation/shallow_water_model/shallow_water_jet_topography.jl

```diff
@@ -0,0 +1,167 @@
+using Oceananigans
+using Oceananigans.Models: ShallowWaterModel
+   
+Nx, Ny = 50, 128
+Lx, Ly = 2π,   10
+H₀     = 10
+α, β   = 0.1,  0.1
+Lb, Hb = α * Ly, β * H₀
+
+f, g, U  = 1, 1, -1.0
+Δη = f * U / g 
+
+grid = RectilinearGrid(size = ( Nx, Ny),
+                          x = (  0, Lx),
+                          y = (-Ly, Ly),
+                   topology = (Periodic, Bounded, Flat))
+
+η(x, y) = Δη * tanh(y)
+b(x, y) = Hb * tanh(y/Lb)
+
+h̄(x, y) = H₀ + η(x, y) + b(x, y)
+ū(x, y) = - U * sech(y)^2
+#ωb(x, y) = - 2 * U * sech(y)^2 * tanh(y)
+
+small_amplitude = 1e-4
+ 
+ uⁱ(x, y) = ū(x, y) + small_amplitude * exp(-y^2) * randn()
+uhⁱ(x, y) = uⁱ(x, y) * h̄(x, y)
+
+ū̄h(x, y) = ū(x, y) * h̄(x, y)
+
+bathymetry(x,y) = -b(x,y)
+
+model = ShallowWaterModel(; grid, coriolis = FPlane(f=f), 
+                gravitational_acceleration = g,
+                               timestepper = :RungeKutta3,
+                        momentum_advection = WENO(),
+                                bathymetry = bathymetry
```

Comment:
what is expected to provide here? here we give `-b`, but what should we give? Why `-b`? is this a convention hidden somewhere? Why not `b` or `-H0+b` or whatever else?

Reply:
I think I'd like to see a sketch  with the free surface and the bathymetry so that I'm clear on what each of these are...

all this confusion of mine started from #2985; see discussion there and, for example, comment https://github.com/CliMA/Oceananigans.jl/pull/2985#issuecomment-1476028985

Reply:
@simone-silvestri , since both @navidcy and I are confused about the convention used here, might you be able to shed some light on the subject?

Reply:
I think the name is confusing because it's vague. Is this the bottom height, where the resting surface displacement is 0? Or is this the thickness of the fluid at rest?

Reply:
I thought it was the bottom height but I could be wrong.

Reply:
I think something like `depth` or `resting_depth` would be less ambiguous maybe

Reply:
Or `resting_layer_height` which is the same thing